src/OrchardCore.Modules/OrchardCore.Tenants/AdminMenu.cs

```diff
@@ -25,7 +25,7 @@ public Task BuildNavigationAsync(string name, NavigationBuilder builder)
             }
 
             // Don't add the menu item on non-default tenants
-            if (_shellSettings.Name != ShellHelper.DefaultShellName)
+            if (!_shellSettings.IsDefaultShell())
```

Comment:
Why do we need this line in this code? If this class is only registered when the tenants module is enabled and the tenant module is only available to the default tenant. Is this something left due to old logic or should it stay as part assurance measure?

Reply:
It is fine. Only the first "default" tenant should be able to create sub-tenants, that's why.

Reply:
Yes I get it. Be the module is only available to host tenant which you can only have one. So I think this code is redundant. I mean even if we remove it this class will only be called by the host tenant. Not sure what I am missing here but

Reply:
It is redundant, though, at some point, it was a necessity. Maybe it could be removed but I would not guess it for now. Leave it there.